ClinGen/clincoded
# Connects to #1092. Connects to #1093. Connects to #1112. Connects to #1056.

**Technical notes:**
1. I pulled Kang's #1114 changes into my branch as I needed to proceed with development and testing. In the process, I have also made additional changes to the schema files as well as the `__init__.py` file. As a result, please consider #1114 being part of my changes and we don't need Kang to create a separate PR for this his branch.
2. Added a note (re: scoring) to the Experimental Assessment panel as an add-on change to the ticket.
3. On the family page, the bulk of the changes apply to the segregation and variant(s) panels. For demographics panel, except for 3 dropdowns, most form fields were simply removed.
4. On the individual page, the bulk of the changes apply to the variant(s) panel, with the addition of score panel.

**Known issues:**
1. The handling of **score status** for Proband currently supports only one score. It needs to be reworked to support an array of scores.
2. The handling of **variant recessive zygosity** for variant only has support for the Proband (with or without family-association) at the moment.

**Steps to test:**
1. Login and add a new PMID from a new or pre-existing GDM.
2. Add a new family. Trying saving the form without filling out the **required** field (e.g. "For Dominant AND Recessive" in _Segregation_ panel) and expect to see error.
3. While on the family page, trying saving the form by entering strings into **Number only** field (e.g. "For Dominant AND Recessive" or "For Recessive Only" in _Segregation_ panel) and expect to see error.
4. While on the family page, in _Segregation_ panel, select either **Yes** or **No** for the "Published Calculated LOD score?" question and expect to see the "Published Calculated LOD score:" or "Estimated LOD score:" questions being shown respectively. And expect to see these 2 input fields accepting decimal numbers.
5. While on the family page, in _Variant(s)_ panel, expect to see 2 new "external" links (variant assessment and curation) associated with each added variant.
6. While on the family page, in _Variant(s)_ panel, expect to see the "Other description" form field present with each added variant.
7. While on the family page, in _Variant(s)_ panel, select "Heterozygous" from the "If Recessive, select variant zygosity" dropdown and expect to see the "Add 2nd variant..." button being shown. Selecting others will remove the button, or the 2nd variant that had been added.
8. Save the family form page and expect to see **NO** errors in the Chrome devtools console.
9. Return to Record Curation page and edit the Individual/Proband page.
10. While on the Proband page, in _Variant(s)_ panel, expect to see added variant(s) from the family page being shown, including the recessive zygosity selection. Also expect to see 3 additional questions in the variant panel.
11. While on the Proband page, expect to see the _Score Proband_ panel at the bottom of the page.
12. After selecting an option from the **required** "sex" form field, select the some options from the "Score Status" dropdown and from the 3 questions in the _Variant(s)_ panel. Save the form and expect to see **NO** errors in the Chrome devtools console.
13. Return to Record Curation page and **add** a new individual (without family association). Select "Yes" from the "Is this Individual a proband" question and expect to see the _Score Proband_ panel being shown at the bottom of the page. Selecting other options would remove the panel.
14. While on the _Individual_ page, expect to be able to repeat the "same" steps/behaviors of adding variant(s) on the Family/Proband pages.
15. On the Family/Individual form pages, expect to see **NO** assessment UIs being shown to prevent users from assessing evidence records (#1112).
16. On the Variant-Curation variation selection box interface, click on the external links among the note text and expect to be taken to the intended sites in new windows/tabs (#1056).
